pennylane/transforms/qcut.py

```diff
@@ -210,3 +213,235 @@ def tape_to_graph(tape: QuantumTape) -> MultiDiGraph:
             order += 1
 
     return graph
+
+
+@dataclass(frozen=True)
+class CutConfig:
+    """
+    Container wrapping partitioning parameters for passing to the automatic cutter.
+    """
+
+    k: int
+    imbalance: float
+    fragment_wires: List[int] = None
+    fragment_gates: List[int] = None
+
+
+def _infer_cut_configs(
+    num_tape_wires,
+    num_tape_gates,
+    max_device_wires,
+    max_device_gates,
+    min_device_wires=None,
+    min_device_gates=None,
+    k_lower=None,
+    k_upper=None,
+    imbalance_tolerance=None,
+    fragment_wires=None,
+    fragment_gates=None,
+):
+    """
+    Helper function for deriving the best default partitioning parameters for the automatic
+    circuit cutter. Arguments can be either user provided or inferred from devices.
+
+    Args:
+        num_tape_wires (int): Number of wires in the circuit tape to be partitioned.
+        num_tape_gates (int): Number of gates in the circuit tape to be partitioned.
+        max_device_wires (int): Number of wires for the largest available device.
+        max_device_gates (int): Maximum allowed circuit depth for the deepest available device.
+        min_device_wires (int): Number of wires for the smallest available device.
+            Optional, defaults to `max_device_wires`.
+        min_device_gates (int): Maximum allowed circuit depth for the shallowest available device.
+            Optional, defaults to `max_device_gates`.
+        k_lower (int): Lowest number of fragments to attempt for the automatic circuit cutter.
+            Optional, defaults to the scenario where each fragment is executed on the largest
+            available device, satisfying both wire and gate constraints.
+            If supplied, can be higher than said default if the the desired k is known.
+        k_upper (int): Largest number of fragments to attempt for the automatic circuit cutter.
+            Optional, defaults to the scenario where each fragment is executed on the smallest
+            available device, satisfying both wire and gate constraints.
+            If supplied, can be higher than said default to encourage exploration.
```

Comment:
The comment above perhaps can help a bit. But the general idea is that, since users don't usually know what k to supply, we need to figure out some sort of sensible defaults. But since we don't know what k is the best before we do any partitioning, we need to try a range of ks and compare later, so what's required is then a sensible default of this range of ks, which is `k_upper` and `k_lower`.

Now `k_lower` is very easy to find out, if your largest device is 40 wires, you'll at least to cut a 100-wire circuit into 3 parts.

But `k_upper` is slightly trickier since there's no strict bound and one can try to cut a circuit into as many fragments as possible. But for a sensible default value, I thought it could be the number of splits needed as if we only use the smallest device in the pool. Or another way to think about this, especially for the case of only one device, this is the number of splits required if we choose to only PARTIALLY occupy wires on the device.